system-addon/lib/FilterAdult.jsm

```diff
@@ -0,0 +1,2974 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+// Keep a Set of adult base domains for lookup (initialized at end of file)
+let gAdultSet;
+
+// Keep a hasher for repeated hashings
+let gCryptoHash = null;
+
+/**
+ * Run some text through md5 and return the base64 result.
+ */
+function md5Hash(text) {
+  // Lazily create a reusable hasher
+  if (gCryptoHash === null) {
+    gCryptoHash = Cc["@mozilla.org/security/hash;1"].createInstance(Ci.nsICryptoHash);
+  }
+
+  gCryptoHash.init(gCryptoHash.MD5);
+
+  // Convert the text to a byte array for hashing
+  gCryptoHash.update(text.split("").map(c => c.charCodeAt(0)), text.length);
+
+  // Request the has result as ASCII base64
+  return gCryptoHash.finish(true);
+}
+
+/**
+ * Filter out any link objects that have a url with an adult base domain.
+ */
+this.filterAdult = function filterAdult(links) {
+  return links.filter(({url}) => {
+    try {
+      const uri = Services.io.newURI(url);
+      return !gAdultSet.has(md5Hash(Services.eTLD.getBaseDomain(uri)));
+    } catch (ex) {
+      return true;
+    }
+  });
+};
+
+this.EXPORTED_SYMBOLS = ["filterAdult"];
+
+// These are md5 hashes of base domains to be filtered out. Originally from:
+// https://hg.mozilla.org/mozilla-central/log/default/browser/base/content/newtab/newTab.inadjacent.json
+gAdultSet = new Set([
```

Comment:
These are a lot of strings. Is there any way to get the list from Firefox itself rather than copy-pasta the entire array of 2900 sites?

Reply:
Those strings dont exist in mozilla-central anymore.